### DISCUSSION TITLE
How to understand schedule with multiple conditions?

### DISCUSSION THREAD

**Discussion by chopeen** [renovatebot/renovate#24407] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

Documentation [#in-repository-schedule-configuration](https://docs.renovatebot.com/key-concepts/scheduling/#in-repository-schedule-configuration) gives the following example to run Renovate outside of common office hours:

```
{
  "schedule": [
    "after 10pm every weekday",
    "before 5am every weekday",
    "every weekend"
  ]
}
```

How should this schedule be interpreted?

1. `"after 10pm every weekday" AND "before 5am every weekday" AND "every weekend"` would mean **never**
2. `"after 10pm every weekday" OR "before 5am every weekday" OR "every weekend"` would mean **24/7**

I think interpretation no. 2 is correct (the conditions are combined with `OR`), but then the example is not very useful.

---

Should it be re-written as follows?

```
{
  "schedule": [
    "after 10pm and before 5am every weekday",
    "every weekend"
  ]
}
```

---

If so, what about `nonOfficeHours`? Should it also be corrected?

https://github.com/renovatebot/renovate/blob/c5aae245caad3288fd4e44519c70e8f5c87e7b12/lib/config/presets/internal/schedule.ts#L8-L12

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment, marked as answer]
If multiple conditions exist, then the logic is OR.

(2) does not mean 24/7 though. After 10pm means 10pm-midnight, before 5am means midnight-5am. A time like 9am does not match either.
